qir/qat/Validator/Tests/Unit/CirqTypeQirsPositive.cpp

```diff
@@ -0,0 +1,173 @@
+  call void @__quantum__qis__cx__body(%Qubit* %1, %Qubit* %0)
+  call void @__quantum__qis__rz__body(double 0x3FE069081E98A785, %Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %1, %Qubit* %0)
+  call void @__quantum__qis__rz__body(double 0xBFE069081E98A785, %Qubit* %0)
+  call void @__quantum__qis__h__body(%Qubit* %0)
+  call void @__quantum__qis__cx__body(%Qubit* %0, %Qubit* %1)
+  call void @__quantum__qis__t__body(%Qubit* %2)
+  call void @__quantum__qis__s__body(%Qubit* %2)
+  call void @__quantum__qis__t__body(%Qubit* %6)
+  call void @__quantum__qis__s__body(%Qubit* %6)
+  call void @__quantum__qis__t__body(%Qubit* %5)
+  call void @__quantum__qis__s__body(%Qubit* %5)
+  call void @__quantum__qis__t__body(%Qubit* %0)
+  call void @__quantum__qis__s__body(%Qubit* %0)
+  call void @__quantum__qis__x__body(%Qubit* %2)
+  call void @__quantum__qis__x__body(%Qubit* %6)
+  call void @__quantum__qis__x__body(%Qubit* %5)
+  call void @__quantum__qis__x__body(%Qubit* %0)
+
+  )script");
+}
+
+TEST(CirqPositive, DISABLED_BernsteinVazirani)
+{
+    // m operations is having the wrong signature
+    expectSuccess("base", R"script(
+  %0 = inttoptr i64 0 to %Qubit*
+  %1 = inttoptr i64 1 to %Qubit*
+  %2 = inttoptr i64 2 to %Qubit*
+  %3 = inttoptr i64 3 to %Qubit*
+  %4 = inttoptr i64 4 to %Qubit*
+  %5 = inttoptr i64 5 to %Qubit*
+  %6 = inttoptr i64 6 to %Qubit*
+  %7 = inttoptr i64 7 to %Qubit*
+  %8 = inttoptr i64 8 to %Qubit*
+  call void @__quantum__qis__x__body(%Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %1)
+  call void @__quantum__qis__h__body(%Qubit* %3)
+  call void @__quantum__qis__h__body(%Qubit* %6)
+  call void @__quantum__qis__h__body(%Qubit* %5)
+  call void @__quantum__qis__h__body(%Qubit* %7)
+  call void @__quantum__qis__h__body(%Qubit* %0)
+  call void @__quantum__qis__h__body(%Qubit* %8)
+  call void @__quantum__qis__h__body(%Qubit* %4)
+  call void @__quantum__qis__h__body(%Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %3)
+  call void @__quantum__qis__h__body(%Qubit* %6)
+  call void @__quantum__qis__h__body(%Qubit* %7)
+  call void @__quantum__qis__h__body(%Qubit* %8)
+  call void @__quantum__qis__x__body(%Qubit* %2)
+  call void @__quantum__qis__cx__body(%Qubit* %1, %Qubit* %2)
+  call void @__quantum__qis__cx__body(%Qubit* %5, %Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %1)
+  call void @__quantum__qis__cx__body(%Qubit* %0, %Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %5)
+  call void @__quantum__qis__cx__body(%Qubit* %4, %Qubit* %2)
+  call void @__quantum__qis__h__body(%Qubit* %0)
+  call void @__quantum__qis__h__body(%Qubit* %4)
+  call void @__quantum__qis__m__body(%Qubit* %1)
+  call void @__quantum__qis__m__body(%Qubit* %3)
+  call void @__quantum__qis__m__body(%Qubit* %6)
+  call void @__quantum__qis__m__body(%Qubit* %5)
+  call void @__quantum__qis__m__body(%Qubit* %7)
+  call void @__quantum__qis__m__body(%Qubit* %0)
```

Comment:
Qubit `%0` is measured twice, the second one can be removed.

Reply:
Where do you see the first measurement?

Reply:
It's measured first on line 161 then again here on line 168. Could be a bug in the high level code that was used to generate this QIR, but given you already said this test case is going to stay disabled then it's not a big deal.

Reply:
Line 161 is an H gate, not measure :) I will leave it as it is as it is as this part is likely to be updated when the converter is done.